pkg/servicetest/testdata/services/docker-compose.solr.yaml

```diff
@@ -9,6 +9,17 @@
 #    so that a file exists with the path ".ddev/solr/conf/solrconfig.xml".
 #
 # Note: The configuration files must be present before running `ddev start`.
```

Comment:
We should add a comment in here that the config is read once for each core defined and if there's a typo or change needed, the volume needs to be deleted i.e. `docker volume rm ddev-myproject_solrdata`, where myproject is the name of site.

Another interesting note for Drupal, the default solr config in search_api_solr is setup for the tarball location of things. Change the defaults to get the core to spin up correctly. `solr.install.dir=/opt/solr` is the correct entry in `.ddev/solr/conf/solrcore.properties`

Reply:
Well, the volume doesn't need to be deleted - the core just needs to be recreated. And, actually, the config just needs to be copied over. I don't see any scripts for that here https://github.com/docker-solr/docker-solr/tree/6f7a6e812247d00f9f3c293993e26d6d041c119e/6.6/scripts

Reply:
@mglaman What are you suggesting as the mitigation since the copy doesn't occur?

Reply:
From the image's docs: https://github.com/docker-solr/docker-solr-examples/tree/master/custom-configset#note

> One note of caution: we've copied the configset from a specific version of Solr. It's possible that a future version will have changes to those config files; it will be up to you to re-generate your config.

🤔

Reply:
I was hoping there was a script in the image we could re-use, I don't have time to dig in at the moment

Reply:
There doesn't seem to be. I did find https://github.com/docker-solr/docker-solr-examples/tree/master/schema-api-startup, which makes me think we should be using schema api. 

For now I wonder if we should just copy/paste that folks who need to make changes to their solr config files need to modify the files inside the docker instance. They won't be overwritten after initial creation.

Reply:
Works for me. If someone is doing this, they're probably an advanced enough user.

> Note: changes to your configuration will not automatically be applied to your Solr core

Reply:
I added an extra note similar to mglaman's wording above.

Reply:
The question is, how can config changes be loaded? Would it be worth noting the steps here?

Reply:
It becomes a question of: is it something to document in DDEV or general official Solr Docker image documentation. I think a fair warning is "good enough"

Reply:
Ultimately, this is an issue with the Solr Docker image, no? It's not DDEV-specific other than that the recommended DDEV config creates a volume. However, not creating a volume means potentially huge reindexing for developers.

Maybe we just need separate examples for various common scenarios so that people have a high chance of getting it right the first time.

Reply:
oh, looks like I am just echoing @mglaman. My review was very slow :)